### ISSUE TITLE
Schema binding by file association with variable works only for active file

### ISSUE THREAD

**Issue by stphnrssr** [redhat-developer/vscode-xml#629]
Hello,
I'm struggling with file association binding of xml schema.
Consider the following data (all files provided in [attached test.zip](https://github.com/redhat-developer/vscode-xml/files/7751649/test.zip)):
- 2 xml files named _data1.xml_ and _data2.xml_
- 2 matching schemas _data1.xsd_ and _data2.xsd_
- binding is provided by file association in VS Code settings as follows:
```
    "xml.fileAssociations": [
        {
            "pattern": "data*.xml",
            "systemId": "${fileBasenameNoExtension}.xsd"
        }
     ]
```
Now, xml validation works fine for each file individually, but problems are systematically reported for other files:

Selecting _data1.xml_, it shows that everything is ok in that one, which is correct. But at the same time, in the _Problems_ panel, an error is reported for _data2.xml_
![image](https://user-images.githubusercontent.com/96472764/146915487-3f53b315-12c1-4e5b-8da6-a7c7e9094909.png)

Same behavior - reversed - when selecting _data2.xml_:
![image](https://user-images.githubusercontent.com/96472764/146915648-9b60c725-ecf9-44a3-85cf-47c6225154a2.png)

And when another file is selected, e.g. _settings.json_, then errors are reported in both xml files:
![image](https://user-images.githubusercontent.com/96472764/146915910-d508e5b5-4185-43da-9ac8-a9797634c4e7.png)

Bottom line, it seems that, when using a variable for file association (such as _${fileBasenameNoExtension}_) **the validation schema is selected on the basis of the active file, and that schema is applied to all validated files**, which produces wrong validation results.

The value of _${fileBasenameNoExtension}_ should return the name of the validated file instead of the name of the active file, shouldn't it?

Note: when using explicit file names for bindings (no variable), everything works fine. Trouble is, my real dataset is much more elaborate than that simple example, and explicit binding of each file is unaffordable... and that issue generates thousands of false alarms.

**Comment by angelozerr** [comment]
Indeed there is a problem with fileBasenameNoExtension. We update fileBasenameNoExtension file associations on XML Language Server side for the activated file editor. With your usecase it's indeed unusable.

We need to improve it but Idon't knowwhen we could do that.
